apps/agentstack-server/src/agentstack_server/service_layer/services/user_feedback.py

```diff
@@ -52,3 +52,20 @@ async def create_user_feedback(
             await uow.user_feedback.create(user_feedback=user_feedback)
             await uow.commit()
             return user_feedback
+
+    async def list_user_feedback(
+        self,
+        *,
+        user: User,
+        provider_id: UUID | None = None,
+        limit: int = 50,
+        after_cursor: UUID | None = None,
+    ) -> tuple[list[UserFeedback], int, bool]:
+        async with self._uow() as uow:
```

Comment:
the service must check that the user owns the provider or is admin:

```suggestion
        user_id = user.id if user.role != UserRole.ADMIN else None
        async with self._uow() as uow:
            # check provider exists and user ownership
            await uow.providers.get(provider_id=provider_id, user_id=user_id)
```